### PR TITLE
swagger add ModelAsString

### PR DESCRIPTION
**Pull request by mjudeikis**
Should we move some of our fields  to `modelAsString` based on https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-enum so we could update them easier if needed in azure specs repo?

For discussion

### CODE REVIEW

File: pkg/swagger/typewalker.go [hunk 1]
```diff
@@ -151,6 +151,18 @@ func (tw *typeWalker) _define(definitions Definitions, t *types.Named) {
 	s.Description = strings.Trim(path[len(path)-2].(*ast.GenDecl).Doc.Text(), "\n")
 	s.Enum = tw.enums[t]
 
+	// Enum extensions allows non-braking api changes
```
Comment (nilsanderselde) [inline pkg/swagger/typewalker.go:154]: *breaking

File: pkg/swagger/types.go [hunk 2]
```diff
@@ -245,7 +245,16 @@ type Schema struct {
 	ExternalDocs         *ExternalDocumentation `json:"externalDocs,omitempty"`
 	Example              interface{}            `json:"example,omitempty"`
 
-	ClientFlatten bool `json:"x-ms-client-flatten,omitempty"`
+	ClientFlatten bool              `json:"x-ms-client-flatten,omitempty"`
+	XMSEnum       *XMSEnumExtension `json:"x-ms-enum,omitempty"`
```
Comment (jim-minter) [inline pkg/swagger/types.go:249]: nit: Enum *Enum `json:"x-ms-enum,omitempty"`

File: pkg/swagger/typewalker.go [hunk 3]
```diff
@@ -151,6 +151,18 @@ func (tw *typeWalker) _define(definitions Definitions, t *types.Named) {
 	s.Description = strings.Trim(path[len(path)-2].(*ast.GenDecl).Doc.Text(), "\n")
 	s.Enum = tw.enums[t]
 
+	// Enum extensions allows non-breaking api changes
+	// https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-enum
+	c := strings.Split(t.String(), ".")
+	name := c[(len(c) - 1)]
+	switch name {
+	case "VMSize":
```
Comment (jim-minter) [inline pkg/swagger/typewalker.go:159]: Don't hard-code this here.  Find a way to pass this in from swagger.go.

File: pkg/swagger/types.go (hunk 2 above)
Comment (mjudeikis) [reply pkg/swagger/types.go:249]: Enum already exists. `XMSEnum` is a separate field.

File: pkg/swagger/typewalker.go (hunk 3 above)
Comment (mjudeikis) [reply pkg/swagger/typewalker.go]: | Suggest we stop generating the 2019-12-31-preview swagger now.

Separate PR

**Comment by mjudeikis** [comment]
| Suggest we stop generating the 2019-12-31-preview swagger now.

Separate PR - https://github.com/Azure/ARO-RP/pull/645

**Comment by mjudeikis** [comment]
I somehow think we should cut new api version for future work?

**Comment by mjudeikis** [comment]
Close in favor of backlog item. Bring back with the item

**Comment by mjudeikis** [comment]
VST 7409284 - closed

**Pull request closed without merging**